### PR TITLE
Speed up nested TaskGoups 

### PR DESCRIPTION
**Pull request by MarosMK**
This PR aims to speed up nested TaskGroups. During the checks in conditions the Task won't have to go into 'depth' to figure out the group_id.
This affects particularly cases where there is deep nesting of TaskGroups.

### CODE REVIEW

**Comment by uranusjr** [comment]
How much speedup does this produce? The `task_group` attribute is public and may potentially change and cause breakage too easily. If the main slowdown is from `child_id` calculation, it may be a better idea to cache at that level instead.

**Comment by MarosMK** [comment]
@uranusjr
The code changes were updated (no caching of group_id) so hopefully there is now less risk at stake and the speedup is the same as before. 
For the time improvement, if we take nested groups of 3 (each task has 3 child tasks) and we go into 'depth' of 6 layers, the dag loading is faster by 7.7s. Or 4 nested groups with 5 layers, it is faster by 3.5s. It depends on how many layers you have and how many task in each layer. 
We do use a lot of nested groups so this would be an improvement.

File: airflow/utils/task_group.py [hunk 1]
```diff
@@ -368,8 +369,10 @@ def child_id(self, label):
         Prefix label with group_id if prefix_group_id is True. Otherwise return the label
         as-is.
         """
-        if self.prefix_group_id and self.group_id:
-            return f"{self.group_id}.{label}"
+        if self.prefix_group_id:
+            group_id = self.group_id
```
Comment (uranusjr) [inline airflow/utils/task_group.py:373]: For nested groups, this implicitly creates a recursion. I wonder if expanding it into a loop would further speed up things significantly.

**Review by uranusjr — APPROVED** [review]
This looks reasonable to me and should be OK to be merged. One more potential path to optimise things though; feel free to investigate in another PR.

File: airflow/utils/task_group.py (hunk 1 above)
Comment (MarosMK) [reply airflow/utils/task_group.py:373]: Yes, that might be true.

**Review by uranusjr — APPROVED** [review]

**Comment by MarosMK** [comment]
@uranusjr, thanks for approving.

Any expected timeline for merging this and including it in a new release?

**Comment by uranusjr** [comment]
You just broke the PR…

**Comment by eladkal** [comment]
> You just broke the PR…

**Review by eladkal — APPROVED** [review]

**Pull request merged**